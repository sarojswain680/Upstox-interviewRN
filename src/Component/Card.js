import get from 'loadsh/get';
import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import usePNLCalculator from '../Hooks/usePNLCalculator';

const Card = item => {
  const stock = get(item, 'item', {});
  const {ltp, quantity, symbol} = stock;
  const {pnl} = usePNLCalculator(stock);

  return (
    <View style={styles.card}>
      <Text style={styles.symbol}>{symbol}</Text>
      <Text style={styles.avgPrice}>{quantity}</Text>
      <View style={styles.rightCard}>
        <Text style={styles.ltp}>LTP: {`₹ ${ltp}`}</Text>
        <Text style={styles.quantity}>P/L: {`₹ ${pnl}`}</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    backgroundColor: '#ffffff',
    padding: 20,
    borderRadius: 10,
    marginVertical: 8,
    marginHorizontal: 16,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  rightCard: {
    marginTop: -55,
  },
  symbol: {
    fontWeight: 'bold',
    alignSelf: 'flex-start',
  },
  avgPrice: {
    marginBottom: 5,
    marginTop: 25,
  },
  ltp: {
    fontWeight: 'bold',
    alignSelf: 'flex-end',
  },
  quantity: {
    fontWeight: 'bold',
    marginTop: 20,
    alignSelf: 'flex-end',
  },
  pl: {
    fontWeight: 'bold',
    alignSelf: 'flex-end',
  },
});

export default Card;
