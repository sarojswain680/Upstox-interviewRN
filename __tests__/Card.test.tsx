import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../src/Component/Card';

describe('Card Component', () => {
  test('renders correctly', () => {
    const item = {
      item: {
        symbol: 'AAPL',
        quantity: 100,
        ltp: 150,
        pnl: 500,
      },
    };

    const tree = renderer.create(<Card {...item} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
