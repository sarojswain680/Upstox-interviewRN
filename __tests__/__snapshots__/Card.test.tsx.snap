// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "borderRadius": 10,
      "elevation": 5,
      "marginHorizontal": 16,
      "marginVertical": 8,
      "padding": 20,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "flex-start",
        "fontWeight": "bold",
      }
    }
  >
    AAPL
  </Text>
  <Text
    style={
      {
        "marginBottom": 5,
        "marginTop": 25,
      }
    }
  >
    100
  </Text>
  <View
    style={
      {
        "marginTop": -55,
      }
    }
  >
    <Text
      style={
        {
          "alignSelf": "flex-end",
          "fontWeight": "bold",
        }
      }
    >
      LTP: 
      ₹ 150
    </Text>
    <Text
      style={
        {
          "alignSelf": "flex-end",
          "fontWeight": "bold",
          "marginTop": 20,
        }
      }
    >
      P/L: 
      ₹ NaN
    </Text>
  </View>
</View>
`;
